packages/node_modules/@webex/internal-plugin-board/src/realtime.js

```diff
@@ -91,19 +91,27 @@ const RealtimeService = WebexPlugin.extend({
   },
 
   createRealtimeChannel(channel) {
-    const requestBindings = [this._boardChannelIdToMercuryBinding(channel.channelId)];
-    const bindingObj = {bindings: requestBindings};
+    let registrationPromise;
 
-    return this.webex.internal.board.register(bindingObj)
-      .then(({webSocketUrl, bindings}) => {
-        this.realtimeChannels.add({
-          channelId: channel.channelId,
-          socketUrl: webSocketUrl,
-          binding: bindings[0]
-        });
+    if (this.config.useFederatedRegistration) {
+      registrationPromise = this.webex.internal.board.registerToChannel(channel)
+        .then(({webSocketUrl, binding}) => ({socketUrl: webSocketUrl, binding}));
+    }
+    else {
+      const requestBindings = [this._boardChannelIdToMercuryBinding(channel.channelId)];
 
-        return this.realtimeChannels.get(channel.channelId);
+      registrationPromise = this.webex.internal.board.register({bindings: requestBindings})
+        .then(({webSocketUrl, bindings}) => ({socketUrl: webSocketUrl, bindings: bindings[0]}));
+    }
+
+    return registrationPromise.then((registration) => {
+      this.realtimeChannels.add({
+        ...registration,
+        channelId: channel.channelId
```

Comment:
Do you not want the `channelId` from `registration` to override the `channelId` from `channel`?

Reply:
The data structure generated by `registrationPromise` doesn't include a `channelId` field.